base/loading.jl

```diff
@@ -636,7 +636,8 @@ function find_all_in_cache_path(pkg::PkgId)
             isfile_casesensitive(filepath) && push!(paths, filepath)
         end
     end
-    return paths
+    p = sortperm(mtime.(paths), rev = true)
+    return paths[p]
```

Comment:
I realised that the `sort!(.. by=..` was calling `mtime` more than once on each file, and given it's expensive it's faster to allocate a sort vector